src/reducers/sources.js

```diff
@@ -419,4 +421,24 @@ export const getSelectedSource = createSelector(
   }
 );
 
+export function getSourceFromPrettySource(
+  state: OuterState,
+  baseSource: Source
+): Source | null {
+  if (!baseSource) {
+    return null;
+  }
+  if (!baseSource.isPrettyPrinted) {
+    return baseSource;
+  }
+
+  const sources = getRelativeSources(state);
+  const id = Object.keys(sources).find(sourceId => {
+    return (
+      getRawSourceURL(sources[sourceId].url) === getRawSourceURL(baseSource.url)
+    );
+  });
+  return id ? sources[id] : null;
+}
```

Comment:
perhaps we move this function definition to right above connect, that way it is clear it is for the source tree and that's it?

Reply:
```diff
diff --git a/src/components/PrimaryPanes/SourcesTree.js b/src/components/PrimaryPanes/SourcesTree.js
index 3a05eea..f11c8b5 100644
--- a/src/components/PrimaryPanes/SourcesTree.js
+++ b/src/components/PrimaryPanes/SourcesTree.js
@@ -368,10 +368,23 @@ class SourcesTree extends Component<Props, State> {
   }
 }

+function getSourceForTree(
+  state: OuterState,
+  baseSource: Source
+): Source | null {
+  if (!baseSource || !baseSource.isPrettyPrinted) {
+    return baseSource;
+  }
+
+  const sources = getRelativeSources(state);
+  const baseSourceUrl = getRawSourceURL(baseSource.url);
+  return getSourceByUrlInSources(state, getRawSourceURL(baseSource.url));
+}
+
 const mapStateToProps = state => {
   return {
-    shownSource: getShownSource(state),
-    selectedSource: getSelectedSource(state),
+    shownSource: getSourceForTree(state, getShownSource(state)),
+    selectedSource: getSourceForTree(state, getSelectedSource(state)),
     debuggeeUrl: getDebuggeeUrl(state),
     expanded: getExpandedState(state),
     projectRoot: getProjectDirectoryRoot(state),

```